src/widgets/fullscreen/fullscreen-widget.ts

```diff
@@ -22,7 +23,13 @@ export interface IFullscreenWidget
   extends ReturnType<typeof FullscreenWidget> {}
 
 export default function createFullscreenWidget(): Promise<IFullscreenWidget> {
-  return createConnection<IFullscreenWidgetEvents>().then(connection =>
-    FullscreenWidget(connection)
-  );
+  let widget: IFullscreenWidget;
+  return createConnection<IFullscreenWidgetEvents>()
+    .then(connection => {
+      widget = FullscreenWidget(connection);
+      connection.sendMessage(PluginMessage.Inited, {
+        plugin_type: PluginType.Fullscreen
+      });
+    })
+    .then(() => widget);
```

Comment:
The `sendMessage` is asynchronous so you want to wait for it before going further with returning the fullscreen widget. Additionally looking at other widget examples you can avoid keeping `widget` as a temporary variable and instead return at the end result of `FullscreenWidget(connection)` directly.

```suggestion
  return createConnection<IFullscreenWidgetEvents>()
    .then(connection => {
      return connection.sendMessage(PluginMessage.Inited, {
        plugin_type: PluginType.Fullscreen
      }).then(() => FullscreenWidget(connection));
    });
```